CONTRIBUTING.md

```diff
@@ -91,21 +77,88 @@ To see test coverage, run `dotnet test` from a console window and you will see t
+Run tests:
+
+```sh
+dotnet test
+```
+
+Check out a new branch, make modifications and push the branch to your fork:
+
+```sh
+$ git checkout -b feature
+# edit files
+$ git commit
+$ git push fork feature
+```
+
+Open a pull request against the main `opentelemetry-dotnet` repo.
+
+### How to Receive Comments
+
+* If the PR is not ready for review, please put `[WIP]` in the title, tag it
+  as `work-in-progress`, or mark it as [`draft`](https://github.blog/2019-02-14-introducing-draft-pull-requests/).
+* Make sure CLA is signed and CI is clear.
+
+### How to Get PRs Merged
+
+A PR is considered to be **ready to merge** when:
+* It has received approval from [Approvers](https://github.com/open-telemetry/community/blob/master/community-membership.md#approver).
+  / [Maintainers](https://github.com/open-telemetry/community/blob/master/community-membership.md#maintainer).
+* Major feedbacks are resolved.
+* It has been open for review for at least one working day. This gives people
+  reasonable time to review.
+* Trivial change (typo, cosmetic, doc, etc.) doesn't have to wait for one day.
+* Urgent fix can take exception as long as it has been actively communicated.
+
+Any Approver / Maintainer can merge the PR once it is **ready to merge**.
+
+## Design Choices
+
+As with other OpenTelemetry clients, opentelemetry-dotnet follows the 
+[opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification).
+
+It's especially valuable to read through the [library guidelines](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/library-guidelines.md).
+
+### Focus on Capabilities, Not Structure Compliance
+
+OpenTelemetry is an evolving specification, one where the desires and
+use cases are clear, but the method to satisfy those uses cases are not.
+
+As such, contributions should provide functionality and behavior that
+conforms to the specification, but the interface and structure is flexible.
+
+It is preferable to have contributions follow the idioms of the language 
+rather than conform to specific API names or argument patterns in the spec.
+
+For a deeper discussion, see: https://github.com/open-telemetry/opentelemetry-specification/issues/165
+
+## Style Guide
 
 This project includes a [`.editorconfig`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/.editorconfig)
 file which is supported by all the IDEs/editor mentioned above.
 It works with the IDE/editor only and does not affect the actual build of the project.
 
 This repository also includes a
-[_test_, _prod_ and _prod.loose_ _stylecop ruleset_](https://github.com/open-telemetry/opentelemetry-dotnet/tree/master/build) files.
-
-These files are used to configure the _StyleCop.Analyzers_ which runs during build. Breaking the rules will result in a broken build.
-
-### Proposing changes
+[test, prod and prod.loose](https://github.com/open-telemetry/opentelemetry-dotnet/tree/master/build) stylecop ruleset files.
```

Comment:
need to confirm if we still have the 3 rulesets.

Reply:
I've optimized the wording here so we don't need to maintain the references.